x-pack/plugins/snapshot_restore/public/application/sections/home/snapshot_list/snapshot_list.tsx

```diff
@@ -41,6 +41,9 @@ export const SnapshotList: React.FunctionComponent<RouteComponentProps<MatchPara
   location: { search },
   history,
 }) => {
+  const unfilteredSnapshotsResponse = useLoadSnapshots(DEFAULT_SNAPSHOT_LIST_PARAMS);
```

Comment:
I believe this bug was introduced when we implemented pagination in https://github.com/elastic/kibana/pull/110266.

I'm a little concerned that we are now making 2 requests to get snapshots. Here, and also on [L58](https://github.com/elastic/kibana/pull/159324/files#diff-796b153bf0363e7144e0d8c29b5345df8b98d45a3e8a75e199a4bf6e1ba85efbL58). Is it possible to avoid this?

Reply:
> I believe this bug was introduced when we implemented pagination in #110266.

That PR was merged into v7.16 and so I pulled [v7.15](https://github.com/elastic/kibana/tree/7.15) locally to test and it seems the bug is present there as well. 🤔 

> I'm a little concerned that we are now making 2 requests to get snapshots. Here, and also on [L58](https://github.com/elastic/kibana/pull/159324/files#diff-796b153bf0363e7144e0d8c29b5345df8b98d45a3e8a75e199a4bf6e1ba85efbL58). Is it possible to avoid this?

Yes, I agree that having 2 requests to Es is not great in terms of performance. The main reason behind this approach is that, based on my understanding, the current implementation filters the snapshots by sending a search request to Es with a query. However, in this way, we cannot get the list of all existing snapshots and so we can't find the correct last successful snapshot since it might not be amongst the filtered ones. That's why I added the second request - without the query - so that we get all existing snapshots and find the last successful one from this list.

The other approach that I had in mind was to send one request without any query and obtain all existing snapshots. Then we can add a function to filter the list based on the filters typed in the search bar. However, at the end this computation might be slower than making a search request to Es with a query. On the other hand, the computation would be performed only when the user adds a filter in the search bar.

What do you think @alisonelizabeth? I'm open to any other ideas or suggestions for improvement.

Reply:
Thanks for the explanation @ElenaStoeva! 

> The other approach that I had in mind was to send one request without any query and obtain all existing snapshots. 

I don't think we should pursue this, as users can potentially have a large number of snapshots (the reasoning why we implemented pagination in https://github.com/elastic/kibana/pull/110266). 

Did you look into if it's possible to fetch only the latest managed snapshot? I think you could leverage the `size` and `order` parameters ([ES docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-snapshot-api.html#get-snapshot-api-query-params)).  However, you would also need the managed repository name and I don't recall if you can specify that you want only successful snapshots returned.

cc @yuliacech as I think she has more familiarity with the pagination work and might have some thoughts too

Reply:
> Did you look into if it's possible to fetch only the latest managed snapshot? I think you could leverage the `size` and `order` parameters ([ES docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-snapshot-api.html#get-snapshot-api-query-params)). However, you would also need the managed repository name and I don't recall if you can specify that you want only successful snapshots returned.

No, I haven't thought about this, but I think this is a great idea. I don't see a query parameter for filtering by state, but since the returned snapshots can be sorted by start time we would just need to return the first successful one (worst case scenario would be having many unsuccessful snapshots). I see the repository name can be returned in the response, so for each returned snapshot we would have to check if its repository is managed.

Reply:
Hi @alisonelizabeth, I added the implementation that we discussed - fetching the last successful managed snapshot only instead of all unfiltered snapshots. If this implementation makes sense, I will add tests for the new function.